packages/near-fast-auth-signer/src/utils/multi-chain/multiChain.ts

```diff
@@ -85,41 +62,30 @@ const signAndSend = async (req: Request): Promise<Response> => {
   } catch (e) {
     return {
       success:      false,
-      errorMessage: e.message
+      errorMessage: e.message,
     };
   }
 };
 
-export const getDerivedAddress = async (
+export const getDerivedBTCAddress = async (
   signerId: string,
-  path: string,
-  chainConfig: ChainConfig,
-  account: Account,
-  contract: ChainSignatureContracts,
-  relayerUrl: string
-) => {
-  let derivedAddress: string;
-
-  switch (chainConfig.type) {
-    case 'EVM':
-      derivedAddress = await EVM.deriveAddress(signerId, path, account, contract, relayerUrl);
-      break;
-    case 'BTC':
-      derivedAddress = (await Bitcoin.deriveAddress(
-        signerId,
-        path,
-        chainConfig.networkType === 'testnet' ? bitcoin.networks.testnet : bitcoin.networks.bitcoin,
-        account,
-        contract,
-        relayerUrl
-      )).address;
-      break;
-    default:
-      throw new Error('Unsupported chain config');
-  }
-
-  return derivedAddress;
-};
+  derivationPath: string,
+  bitcoinNetwork: 'mainnet' | 'testnet',
+  nearNetworkId: 'mainnet' | 'testnet',
+  multichainContractId: ChainSignatureContracts
+) => (
+  await Bitcoin.deriveAddress(
+    signerId,
+    derivationPath,
+    bitcoinNetwork === 'testnet'
+      ? bitcoin.networks.testnet
+      : bitcoin.networks.bitcoin,
+    nearNetworkId,
+    multichainContractId
+  )
+).address;
+
+export const getDerivedEVMAddress = EVM.deriveAddress;
```

Comment:
For this static methods, isn't better simply expose the class? Doing what we are doing here it's just adding more code to `maintain`, like the getDerivedBTCAddress. 

If the function or static method interface changes we will have to update it in two places. I think it's better simply expose the class method

We can have a chat on it. Because I'm not 100% happy with making most of the class method static, then we lose the class purpose.